src/fit.py

```diff
@@ -1,23 +1,48 @@
 """Contains the necessary tools to fit a dataset."""
 
+import os.path
+import typing as tp
+
 import matplotlib.pyplot as plt
 import numpy as np
 from scipy.optimize import curve_fit
 
 # We need the value of the earth gravity to determine the initial velocity and initial angle
 from generate import EARTH_GRAVITY
 
+REPO_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+DATA_DIR = os.path.join(REPO_DIR, "data")
+PLOT_DIR = os.path.join(REPO_DIR, "plots")
+
 # Step 1: Determine the starting velocity and starting angle of the test dataset `data/A-test.txt`.
 
 ## 1. Load data file using https://numpy.org/doc/stable/reference/generated/numpy.loadtxt.html#numpy-loadtxt .
 ##    The first column corresponds to the abscissa x and the second column to the ordinate y.
+# See the main section below.
+
 
 ## 2. Define the model: Look to the example on SciPy (the link below)
+def model(x, a, b):
+    return a * x + b * x**2
+
 
 ## 3. Fit using https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.curve_fit.html#curve-fit
+def fit(x: np.ndarray, y: np.ndarray) -> tp.Tuple[float, float]:
+    sol = curve_fit(model, xdata=x, ydata=y)
```

Comment:
this is of course just fine - but using `popt, pcov = curve_fit(func, xdata, ydata)` as [here](https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.curve_fit.html#curve-fit) is much more expressive